features/step_definitions/branch_steps.rb

```diff
@@ -80,7 +80,16 @@
 end
 
 
+Given(/^I am on the "(.+?)" branch with "(.+?)" as the previous Git branch/) do |current_branch, previous_branch|
+  step %(I am on the "#{previous_branch}" branch)
+  step %(I am on the "#{current_branch}" branch)
```

Comment:
This is fine, but I think it actually obfuscates what you are doing here and just performing the two checkouts here would make it simpler.